src/AutoMapper/QueryableExtensions/ProjectionBuilder.cs

```diff
@@ -243,16 +297,43 @@ public override QueryExpressions GetSubQueryExpression(ProjectionBuilder builder
             return new(Lambda(projection, secondParameter), Lambda(letClause, instanceParameter));
             void ReplaceSubQueries()
             {
-                foreach(var letMapInfo in letMapInfos)
+                for (int i = 0; i < letMapInfos.Length; i++)
                 {
+                    var letMapInfo = letMapInfos[i];
                     var letProperty = letType.GetProperty(letMapInfo.Property.Name);
                     var letPropertyMap = letTypeMap.FindOrCreatePropertyMapFor(letProperty, letMapInfo.Property.Type);
-                    letPropertyMap.MapFrom(Lambda(letMapInfo.LetExpression.ReplaceParameters(letMapInfo.MapFromSource), secondParameter));
+
+                    var source = letMapInfo.MapFromSource;
+                    var extractor = new ParameterExpressionExtractorVisitor();
+                    extractor.Visit(source);
+
+                    var type = letMapInfo.LetExpression.Parameters.First().Type;
+                    if (source is ConditionalExpression condition) 
+                    {
+                        source = Condition(condition.Test, letMapInfo.LetExpression.ReplaceParameters(condition.IfTrue), letMapInfo.Marker);
+                    }
+                    else if (extractor.Parameter.Type != type && source.Type != type)
+                    {
+                        source = Condition(TypeIs(source, type), letMapInfo.LetExpression.ReplaceParameters(Convert(source, type)), letMapInfo.Marker);
+                        for (int j = i+1; j < letMapInfos.Length; j++)
+                        {
+                            if (letMapInfos[j].MapFromSource.Type == letMapInfo.Marker.Type)
+                            {
+                                letMapInfos[j] = letMapInfos[j] with { MapFromSource = source };
```

Comment:
The chaining should happen in `GetSourceExpression`, not here.

Reply:
I changed it on purpose as a resultig expression of one let property can be a source of another property. im not sure how to  code this recursion when the code is in  GetSourceExpression

Reply:
understood

Reply:
I was able to simplify code and reuse it some more let me know what you think...